frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -2847,6 +2847,18 @@ const IR::Node* TypeInference::postorder(IR::ConstructorCallExpression* expressi
     return expression;
 }
 
+void TypeInference::explodeStructType(const IR::Type_StructLike* structType, IR::Type_Tuple *tuple) {
```

Comment:
Maybe you can make this a static file-level method (with no class), since it does not depend on "this".
I would call it "convertStructToTuple"